pilot/pkg/model/typed_xds_cache.go

```diff
@@ -103,108 +103,145 @@ type typedXdsCache[K comparable] interface {
 	Keys() []K
 	// Snapshot returns a snapshot of all keys and values. This is for testing/debug only
 	Snapshot() []*discovery.Resource
+	// closes cache, frees all resources
+	Close()
 }
 
 // newTypedXdsCache returns an instance of a cache.
 func newTypedXdsCache[K comparable]() typedXdsCache[K] {
 	cache := &lruCache[K]{
 		enableAssertions: features.EnableUnsafeAssertions,
-		configIndex:      map[ConfigHash]sets.Set[K]{},
-		evictQueue:       make([]evictKeyConfigs[K], 0, 1000),
+		configIndex:      xsync.NewMapOf[ConfigHash, sets.Set[K]](),
 	}
 	cache.store = newLru(cache.onEvict)
+	cache.token.Store(0)
 	return cache
 }
 
-type evictKeyConfigs[K comparable] struct {
-	key              K
-	dependentConfigs []ConfigHash
-}
-
 type lruCache[K comparable] struct {
 	enableAssertions bool
-	store            simplelru.LRUCache[K, cacheValue]
+	store            otter.Cache[K, cacheValue]
 	// token stores the latest token of the store, used to prevent stale data overwrite.
 	// It is refreshed when Clear or ClearAll are called
-	token       CacheToken
-	mu          sync.RWMutex
-	configIndex map[ConfigHash]sets.Set[K]
-
-	evictQueue []evictKeyConfigs[K]
-
-	// mark whether a key is evicted on Clear call, passively.
-	evictedOnClear bool
+	token atomic.Uint64
+	// we need this mutex to protect operations which require unsafe modifications
+	mu          xsync.RBMutex
```

Comment:
Not sure if this is right, seems we need to protect both configIndex and store to make they work atomic